### ISSUE TITLE
Additional core block exclusions on aria-hidden

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#567]
## Is your feature request related to a problem? Please describe.

There are some correct uses of aria-hidden related to core blocks that it would be good for us to exclude from flagging an aria-hidden warning because we know that they are correct.

## Describe the solution you'd like
<img width="985" alt="Screenshot 2024-03-30 at 9 38 13 PM" src="https://github.com/equalizedigital/accessibility-checker/assets/13153456/e86ef8a7-dc3b-4e06-b360-88a9b7d07af1">

Here are the code snippets:
`<span aria-hidden="true" class="wp-block-cover__background has-background-dim"></span>`

`<span class="wp-block-post-featured-image__overlay has-background-dim has-background-dim-10 has-contrast-2-background-color" style="border-radius:5px" aria-hidden="true"></span>`

## Why do you think this feature is something we should consider for this plugin?

We should avoid as many false positives as we can and make our rules the best ruleset for WordPress. 🙂

**Comment by pattonwebz** [comment]
@amberhinds does the `.has-background-dim` class matter here on either of these 2 pieces of block markup when it comes to determining if they are valid uses of aria-hidden or not?

**Comment by amberhinds** [comment]
I think you can use that as the attribute to ignore, probably. It's the span that overlays images to ensure text on them is readable on the cover block. You can see one on my website here: https://amberhinds.com/about/

<img width="1786" alt="Screenshot 2024-04-01 at 9 14 52 AM" src="https://github.com/equalizedigital/accessibility-checker/assets/13153456/41192407-cc6f-4866-b8ad-2544ad6be774">

**Comment by SteveJonesDev** [comment]
Issue moved to our internal system.

**Comment by SteveJonesDev** [comment]
Since the rules can also be used outside of WordPress as a standalone package, we want to avoid adding WordPress-specific class names directly into the JS check logic.

The approach we'd like to take here is adding an `exclusions` key to the rule's PHP config array (e.g. in `AriaHiddenRule.php`). The JS check would read those selectors dynamically rather than having them hardcoded. WordPress-specific patterns live in the plugin layer where they belong, and the generic rule stays clean.

Example:

```php
'exclusions' => [
    '.wp-block-cover__background.has-background-dim',
    '.wp-block-post-featured-image__overlay.has-background-dim',
],
```

As part of this work we'd also move the existing `wp-block-spacer` exclusion out of the JS and into this config, since it has the same problem.

This approach requires building out the exclusions config support before the specific patterns can be added, so the scope is a bit larger than a quick class check, but it feels like the right foundation going forward.
